### ISSUE TITLE
Symlinks are problematic (index.html)

### ISSUE THREAD

**Issue by fuhrmanator** [webpro/reveal-md#330]
I found an inconsistent behavior with reveal-md in in this [specific deployment scenario](https://github.com/martinmurphy/slidestemplate/blob/master/.travis.yml).  

The final step of the command `reveal-md slides.md --static docs` results in a symlink being created, e.g., `∞ docs\slides.html → docs\index.html`.

1. Under Windows 10 (in VS Code) it appears this step fails if administrator privileges aren't given:
```
[Error: EPERM: operation not permitted, symlink 'slides.html' -> 'C:\Users\blah\docs\index.html']
```
2. If the symlink is created in Windows but then pushed to a GitHub repo (depending on how git is configured?), the `docs/index.html` in the remote repo (GitHub) won't be a symlink but a regular file. So, if Travis-CI is used to then again deploy the file to GitHub pages, for example, and an index.html **file** (not symlink) exists, the command `reveal-md slides.md --static docs` fails the symlink step (it doesn't work, and the former version of `docs/index.html` remains unchanged). No error is reported. You can repeat this behavior locally by doing `touch index.html` on an empty `docs` directory. 

Because of this behavior, debugging a Travis-ci deployment (that doesn't appear to update anything) is super problematic! 

I think it would be awesome if there was a choice to use symlinks in the command (it's not a problem to me if my index.html is a copy of the slides.html, or even if slides.html was renamed). Doing an `rm` on the `index.html` might make sense early on in the generate process, or generating an error if the `index.html` file exists and is not already a symlink.

**Comment by webpro** [comment]
I guess a check to see if this `index.html` exists before creating this symlink would suffice, right? And a warning if it can't be created.

**Comment by fuhrmanator** [comment]
@webpro some kind of warning as you suggest, including a notification that a symlink is being used sounds good to me. 

I personally have lost much time in my 30+ years of coding because of troubles with symlinks, especially in Git, and especially in cross-platform projects. So I am less in favor of using them as a rule. Making a copy of `index.html` is just some extra bytes and not the [accidental complexity of a symlink (the struggle is real!)](https://stackoverflow.com/q/5917249/1168342).

**Comment by webpro** [comment]
Should be fixed in [v4.1.1](https://github.com/webpro/reveal-md/releases/tag/4.1.1) - it's a "safe copy" (don't copy if file exists).

**Comment by fuhrmanator** [comment]
> (don't copy if file exists)

Does it result in an error then? Using the Travis-CI scenario as above, if it won't overwrite the `index.html` then (if I'm understanding correctly), the same mystery will happen. 

I think it would be OK to overwrite an existing index.html file when using the `--static` option.

**Comment by webpro** [comment]
1. reveal-md no longer creates a symlink. That should fix issues re. admin privileges and accidental complexity.
1. I now read the initial, second issue better, sorry for misunderstanding.

I've just released [v4.1.2](https://github.com/webpro/reveal-md/releases/tag/4.1.2) which copies (and overwrites) the file. A copy/overwrite indeed should be fine when generating such static files.
